File: backend/src/main/java/it/unical/inf/ea/backend/dto/CartItemDTO.java
package it.unical.inf.ea.backend.dto;

import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CartItemDTO {

    @NotNull
    private String id;

    @NotNull
    private Long productId;

    @NotNull
    private int quantity;
}

File: frontend/app/src/main/java/com/android/frontend/controller/models/CartItemDTO.kt
package com.android.frontend.controller.models

import java.math.BigDecimal

data class CartItemDTO(
    val id: String,
    val productId: String,
    val quantity: Int,
    val productName: String,
    val productPrice: BigDecimal,
    val deliveryPrice: BigDecimal
)
